models/stg_google_ads__campaign_history.sql

```diff
@@ -34,6 +34,15 @@ final as (
         serving_status,
         status,
         tracking_url_template,
+        --Adding final_url_suffix because sometimes the UTMs are only configured in the final_url_suffix than the tracking template
+        final_url_suffix,
+        --Extracting UTMs for campaign report's use 
+        coalesce( {{ dbt_utils.get_url_parameter('tracking_url_template', 'utm_source') }} ,
+                  {{ dbt_utils.get_url_parameter('final_url_suffix', 'utm_source') }} ) as utm_source,
```

Comment:
I noticed you included a coalesce here for `tracking_url_template` if `final_url_suffix` is not populated. However, when checking my data I don't see `tracking_url_template` contain any utm parameters. Is this coalesce needed?

Reply:
My understanding is that `tracking_url_template` could hold UTM if hard coded into the template. This isn't encouraged in the Google documentation, but it is a possibility. I would personally flip the order so coalesce would pick from `final_url_suffix` first if it exists then check the template as `final_url_suffix` is the main specified entry for tracking parameters to be placed.

Reply:
This is great input and thanks for confirming the behavior on your end! I agree that flipping would likely be the best option.